### PR TITLE
Adds Pill component docs to site

### PR DESCRIPTION
**Pull request by james-nash**
Adds the Pill component docs from the completed Word template to the site

### Notes for reviewers:

* You can preview the page here: https://saltdesignsystem-git-add-pill-docs-joshwooding.vercel.app/salt/components/pill/examples
* This PR only converts the text content from Word to MDX. The content has already been reviewed in Word, so you just need to focus your review on whether or not it's been added to the site correctly.
* The example code still needs to be migrated out of Storybook, so you will just see a dummy example on the page for now (but the titles & descriptions for each example should already be correct)
* The props table will appear empty because it's expecting to find the code in the `core` package, but this component is still in `labs`. I figured it's better to do it this way around rather than setting it to `labs` now and risking forgetting to update that before this page goes live to consumers. (I did test it with labs though and the props appears correctly, so this _will_ work when the time comes)
* I'm aware that the Accessibility tab needs some spacing between the tabs and the text. However, this is a bug in the page styling and nothing to do with this component's content. I'll therefore address that in a separate PR

### CODE REVIEW

File: site/docs/components/pill/usage.mdx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+#### When not to use Pill
+
+- When the content is a system message and cannot be interacted with. Use Badge instead.
+
+### Import
+
+{/* Update the text and code snippet below as needed: */}
+
+To import Component Name from the core Salt package, use:
```
Comment (origami-z) [inline site/docs/components/pill/usage.mdx:28]: "Component Name"?

File: site/docs/components/pill/index.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+data:
+  description: Pill is a small visual element that contains versatile content. You can use Pill to label, tag or categorize content. With Pill, users can trigger actions, make selections or filter results. Actions that may change depending on the context, and are not fixed, are better represented by Pill than Button.
+
+  # Fill in the info from the content template's "Metadata" table below.
+  # To omit optional items, comment them out with #
+  sourceCodeUrl: "https://github.com/jpmorganchase/salt-ds/blob/pill-next/packages/core/src/pill/"
+  package:
+    name: "@salt-ds/core"
+  alsoKnownAs: ["Tag", "Chip", "Badge"]
+  relatedComponents: [
```
Comment (origami-z) [inline site/docs/components/pill/index.mdx:12]: Should "Contains" display nothing or the text should not be there as well?

File: site/docs/components/pill/usage.mdx (hunk 1 above)
Comment (james-nash) [reply site/docs/components/pill/usage.mdx:28]: Oops! That's a leftover from the template I forgot to replace. Fixing that now...

File: site/docs/components/pill/index.mdx (hunk 2 above)
Comment (james-nash) [reply site/docs/components/pill/index.mdx:12]: The heading should disappear. It's a bug in the component page layout. I'm working on a separate PR to fix that and a few other minor issues I've found on the component page layout.

**Review by feliciackh — APPROVED** [review]

**Comment by james-nash** [comment]
Replaced by #2300

**Pull request closed without merging**